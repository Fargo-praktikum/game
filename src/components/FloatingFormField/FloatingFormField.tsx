import { useField } from "formik";
import React from "react";
import { TextBox } from "../TextBox";
import { FloatingFormFieldProps } from "./types";

export const FloatingFormField = (props: FloatingFormFieldProps): JSX.Element => {

    const [field, meta] = useField(props);

    return (
        <div className="floating-label-form__field">
            <TextBox
                type={props.type ?? "text"}
                placeholder={props.placeholder ?? props.label}
                className="floating-label-form__text-box"
                {...field}
            />
            <label className="floating-label-form__label">{props.label}</label>
            {meta.touched && meta.error ? (
                <div className="floating-label-form__error-message">{meta.error}</div>
            ) : null}
        </div>
    );
};
