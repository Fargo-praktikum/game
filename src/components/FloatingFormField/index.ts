import { FloatingFormField } from "./FloatingFormField";

export { FloatingFormField };
