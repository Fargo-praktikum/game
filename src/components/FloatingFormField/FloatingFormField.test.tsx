import React from "react";
import renderer from "react-test-renderer";
import { Form, Formik } from "formik";

import { FloatingFormField } from "./FloatingFormField";


describe("Проверка FloatingFormField с помощью snapshot", () => {
    it("корректная отрисовка", () => {
        const tree = renderer.create(
            <Formik<{test: string}>
                initialValues={{ test: "" }}
                onSubmit={() => {return;}}
            >
                {() => (
                    <Form>
                        <FloatingFormField
                            name="login"
                            label="Логин"
                        />
                    </Form>
                )}
            </Formik>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("все три свойства", () => {
        const tree = renderer.create(
            <Formik<{test: string}>
                initialValues={{ test: "" }}
                onSubmit={() => {return;}}
            >
                {() => (
                    <Form>
                        <FloatingFormField
                            name="password"
                            label="Пароль"
                            type="password"
                        />
                    </Form>
                )}
            </Formik>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});



