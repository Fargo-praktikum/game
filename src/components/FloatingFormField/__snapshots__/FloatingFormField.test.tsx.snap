// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка FloatingFormField с помощью snapshot все три свойства 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="floating-label-form__field"
  >
    <input
      className="text-box floating-label-form__text-box"
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Пароль"
      type="password"
    />
    <label
      className="floating-label-form__label"
    >
      Пароль
    </label>
  </div>
</form>
`;

exports[`Проверка FloatingFormField с помощью snapshot корректная отрисовка 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="floating-label-form__field"
  >
    <input
      className="text-box floating-label-form__text-box"
      name="login"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Логин"
      type="text"
    />
    <label
      className="floating-label-form__label"
    >
      Логин
    </label>
  </div>
</form>
`;
